.github/workflows/release.yml

```diff
@@ -20,30 +20,26 @@ jobs:
         with:
           egress-policy: audit
 
-      - name: Checkout repo
+      - name: Checkout Repo
         uses: actions/checkout@v4
         with:
           fetch-depth: 0
 
-      - name: Build
-        uses: ./.github/actions/build
-
-      - name: Create or update branch
+      - name: Create or Update Branch
         env:
           BRANCH_NAME: ${{ github.event.inputs.branch-name }}
         run: node ./build/release-version.js
 
+      - name: Build and Run Unit Tests
+        uses: ./.github/actions/build
+
```

Comment:
build only after the `./build/release-version.js` script has created and checked out the branch.